### ISSUE TITLE
Autocomplete for assets directory input

### ISSUE THREAD

**Issue by ax-vasquez** [nuggylib/minecraft-asset-reader#40]
# Description

As of now, the user has to paste in the full-path to the assets directory. We should add autocomplete to make it easier for the user to provide their assets directory.

**Comment by armstrongl** [comment]
It looks like we'll be able to resolve the pain point of having to manually enter in the assets directory by incorporating an autocomplete library. The [omelette](https://www.npmjs.com/package/omelette) tool seems to have a good solution, and it supports pulling autocompletions from `fs.readdirSync`.

The implementation might look something like this:
```javascript

omelette`
import * as omelette from 'omelette'
    /|\
        ${() => fs.readdirSync("some/dir/path/")}
`init()
```

**Comment by armstrongl** [comment]
Steps:
1. Add a utility to determine the user's operating system using `process.platform` (e.g., `var isWin = process.platform === "win32"`) or the `os` module (e.g., `console.log(os.platform()); // "win32"`). 
2. Check the default Minecraft directory for the user's operating system. 
3. If it exists, populate the assets directory.
    - If it's only a `.jar`, extract the `.jar`.
4. If it does not, prompt for the non-default directory.
    - If it's only a `.jar`, extract the `.jar`.

**Comment by armstrongl** [comment]
How to detect the current operating system:
```javascript
switch (os.type())
	case 'Linux'
		// Logic for Linux
	case 'Darwin'
		// Logic for Darwin (Mac OS X)
	case 'Windows_NT'
		// Logic for Windows
	default
		// Handle unknown operating system

```

How check if a directory exists:
```javascript
const fs = require("fs")
const linuxDefaultDir = "/home/$USER/.minecraft/"

try {
	if (fs.existsSync(linuxDefaultDir)) {
		console.log("Directory exists.")
		defaultDir = linuxDefaultDir
	} else {
		console.log(`Directory ${linuxDefaultDir} does not exist.`)
		// Prompt for non-default dir
	} catch(e) {
		console.log("An error occurred while checking for if the default directory exists.")
	}
}
```

How to check if a file exists:
```javascript
// You can check if a file exists the same way you check if a directory exists; with fs.existsSync().

const fs = require('fs')
const path = './file.txt'

try {
	  if (fs.existsSync(path)) {
		// file exists
	  }
	} catch(err) {
		console.error(err)
	}

```

Execute command via node.js:
```javascript
const { exec } = require("child_process")

exec("ls -la", (error, stdout, stderr) => {
	if (error) {
		console.log(`error: ${error.message}`)
		return
	}
	if (stderr) {
		console.log(`stderr: ${stderr}`)
		return
	}
	console.log(`stdout: ${stdout}`)
})
```

How to extract a `.jar` file:
```javascript
const { exec } = require("child_process")

let extractJarCmd = `jar xvf ${pathToJar}`

exec(extractJarCmd, (error, stdout, stderr) => {
	if (error) {
		console.log(`error: ${error.message}`)
		return
	}
	if (stderr) {
		console.log(`stderr: ${stderr}`)
		return
	}
	console.log(`stdout: ${stdout}`)
})

```


All together now:
```javascript
const fs = require("fs")

const linuxDefaultDir = "/home/$USER/.minecraft/versions/"
const darwinDefaultDir = "~/Library/Application Support/minecraft/versions/"
const winDefaultDir = "%appdata%\.minecraft\versions\"

const currentOs = os.type()
let defaultDir = ""
// TODO: Prompt for version
let minecraftVersion = "1.12.2"

const linuxJar = `/home/$USER/.minecraft/versions/${minecraftVersion}/${minecraftVersion}.jar`
const darwinJar = `~/Library/Application Support/minecraft/versions/${minecraftVersion}/${minecraftVersion}.jar`
const winJar = `%appdata%\.minecraft\versions\\${minecraftVersion}\\${minecraftVersion}.jar`



function checkForJar(currentOs) {

	switch(currentOS)
		case("Linux")
			try {
				if (fs.existsSync(linuxJar)) {
					console.log("Jar exists.")
					console.log("Extracting jar...")
					extractJar(linuxJar)
					
				} else {
					console.log(`Jar does not exist.`)

				} catch(e) {
					console.log("An error occurred while checking for jar file.")
				}
			}
		case("Darwin")
			try {
				if (fs.existsSync(darwinJar)) {
					console.log("Jar exists.")
					console.log("Extracting jar...")
					extractJar(darwinJar)
					
				} else {
					console.log(`Jar does not exist.`)

				} catch(e) {
					console.log("An error occurred while checking for jar file.")
				}
			}
		case("Windows_NT")
			try {
				if (fs.existsSync(winJar)) {
					console.log("Jar exists.")
					console.log("Extracting jar...")
					extractJar(winJar)
					
				} else {
					console.log(`Jar does not exist.`)
				} catch(e) {
					console.log("An error occurred while checking for jar file.")
				}
			}
}


function extractJar(path) {
	let extractJarCmd = `jar xvf ${path}`

	exec(extractJarCmd, (error, stdout, stderr) => {
		if (error) {
			console.log(`error: ${error.message}`)
			return
		}
		if (stderr) {
			console.log(`stderr: ${stderr}`)
			return
		}
		console.log(`stdout: ${stdout}`)
	})
}

switch (os.type())
	case 'Linux'
		try {
			if (fs.existsSync(linuxDefaultDir)) {
				console.log("Directory exists.")
				defaultDir = linuxDefaultDir
			} else {
				console.log(`Directory ${linuxDefaultDir} does not exist.`)
				// Prompt for non-default dir
			} catch(e) {
				console.log("An error occurred while checking for if the default directory exists.")
			}
		}
	case 'Darwin'
		try {
			if (fs.existsSync(darwinDefaultDir)) {
				console.log("Directory exists.")
				defaultDir = darwinDefaultDir
			} else {
				console.log(`Directory ${darwinDefaultDir} does not exist.`)
				// Prompt for non-default dir
			} catch(e) {
				console.log("An error occurred while checking for if the default directory exists.")
			}
		}
	case 'Windows_NT'
		try {
			if (fs.existsSync(winDefaultDir)) {
				console.log("Directory exists.")
				defaultDir = winDefaultDir
			} else {
				console.log(`Directory ${winDefaultDir} does not exist.`)
				// Prompt for non-default dir
			} catch(e) {
				console.log("An error occurred while checking for if the default directory exists.")
			}
		}
	default
		// prompt for dir because OS detection failed
```

**Comment by armstrongl** [comment]
Changes added so far:
- Added a `utils` directory in `src`. 
- Added a `checkForAssets` function that does the following:
    - Gets the current operating system. 
    - Checks if the default Minecraft assets directory exists (differs based on the operating system)
        - If the assets directory doesn't exist, it detects and extracts the `.jar`, then gets the assets directory.
    - Sets the assets directory and returns it. 
- Added a new option that the user can select to use the default Minecraft directory for their operating system.

**Comment by armstrongl** [comment]
Closed in favor of #73
